package/scripts/cg_limit.sh

```diff
@@ -0,0 +1,256 @@
+        packets, default : $MARKID
+\e[1m-x\e[0m       Removes all limits
+\e[1m-h\e[0m       Shows this help
+UNITS:\n
+        bit or a bare number Bits per second
+        kbit   Kilobits per second
+        mbit   Megabits per second
+        gbit   Gigabits per second
+        tbit   Terabits per second
+        bps    Bytes per second
+        kbps   Kilobytes per second
+        mbps   Megabytes per second
+        gbps   Gigabytes per second
+        tbps   Terabytes per second
+        s, sec or secs Whole seconds
+        ms, msec or msecs Milliseconds
+        us, usec, usecs or a bare number Microseconds.
+For the download rate, because of iptables's hashlimit limitations
+only bytes units are valid and the maximum value is 400mbps."
+}
+
+# Helper functions to handle unit conversion
+toBytes() {
+  echo $1 | awk \
+          'BEGIN{IGNORECASE = 1}
+          /[0-9]$/{print $1};
+          /kbps?$/{printf "%ukb\n", $1; exit 0};
+          /mbps?$/{printf "%umb\n", $1; exit 0};
+          /gbps?$/{printf "%ugb\n", $1; exit 0};
+          /bps?$/{printf "%ub\n", $1};'
+}
+
+if [[ $EUID -ne 0 ]]; then
+   echo "This script must be run as root" 1>&2
+   exit 1
+fi
+
+# Default highest values
+U_LIMIT=400gbps
+D_LIMIT=400mb
+DELAY_LIMIT=0
+
+while getopts ":hm:d:u:l:c:a:b:i:x" option
+do
+  case $option in
+  m)
+    M_FLAG=1
+    M_LIMIT=$OPTARG
+    ;;
+	u)
+		U_FLAG=1
+    U_LIMIT=$OPTARG
+		;;
+	d)
+		D_FLAG=1
+    if [[ $OPTARG =~ .*bit ]]; then
+      echo "Please provide the download rate in bytes"
+      exit 1
+    fi
+    D_LIMIT=$(toBytes $OPTARG)
+		;;
+  l)
+    DELAY_FLAG=1
+    DELAY_LIMIT=$OPTARG
+    ;;
+  c)
+    CGCLASSID=$OPTARG
+    ;;
+  a)
+    MARKID=$OPTARG
+    ;;
+  b)
+    UNAME=$OPTARG
+    ;;
+  i)
+    DEV=$OPTARG
+    ;;
+  h)
+    display_help
+    exit 2
+    ;;
+  x)
+    delete_limit_rate_latency_egress
+    delete_limit_rate_ingress $CGCLASSID $MARKID
+    delete_cgroup_net $CGNAME
+    exit 0
+    ;;
+  :)
+    echo "The '$OPTARG' options requires an argument" >&2
+    exit 1
+    ;;
+  \?)
+    echo "'$OPTARG' : invalid option check help page"
+    exit 1
+    ;;
+  esac
+done
+
+if [[ -z "$UNAME" ]]; then
+  echo "You have to provide a username (-b option)"
+  exit 1
+fi
+
+if [[ "$M_FLAG" -eq "1" ]]; then
+  init_cgroup_mem $CGNAME $M_LIMIT
+  if [ $? -ne 0 ]; then
+    delete_cgroup_mem $CGNAME
+    exit 1
+  fi
+  cgclassify -g memory:/$CGNAME $$
+fi
+if [[ "$U_FLAG" -eq "1" ]] || [[ "$DELAY_FLAG" -eq "1" ]] || [[ "$D_FLAG" -eq "1" ]]; then
+  init_cgroup_net $CGNAME $CGCLASSID
+  limit_rate_latency_egress $DEV $U_LIMIT $DELAY_LIMIT $MARKID
+  limit_rate_ingress $D_LIMIT $CGCLASSID $MARKID
+  if [ $? -ne 0 ]; then
+    delete_limit_rate_latency_egress
+    delete_limit_rate_ingress $CGCLASSID $MARKID
+    delete_cgroup_net $CGNAME
+    exit 1
+  fi
+  cgclassify -g net_cls:/$CGNAME $$
+fi
+
+# now executing the program
+shift "$((OPTIND - 1))"
+sudo -H -u $UNAME $@
```

Comment:
Nice then if the problem is resolved. I don't see any reasons that would cause conflicts. The best way to prove this would be to test this script on multiple distributions.